### PR TITLE
[ASF] ASF Publish

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

- Add release script
- Add release guide
- Add license header in some files

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

The release script has not been tested yet.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -51,7 +51,7 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          asset_paths: '["./kyuubi-*tar.gz"]'
+          asset_paths: '["./kyuubi-*tgz"]'
       - name: Update Release Note
```
Comment (yaooqinn) [inline .github/workflows/release.yml:55]: does this action actually be used？

Comment (pan3793) [reply .github/workflows/release.yml:55]: I'm going to remove this job. There are still lots of works to do for apache release.

Comment (yaooqinn) [reply .github/workflows/release.yml:55]: thank you

File: build/dist [hunk 2]
```diff
@@ -145,20 +145,19 @@ HIVE_VERSION=$("$MVN" help:evaluate -Dexpression=hive.version $@ 2>/dev/null\
 
 echo "Building Kyuubi package of version $VERSION against Spark version - $SPARK_VERSION"
 
+SUFFIX="-$NAME"
```
Comment (yaooqinn) [inline build/dist:148]: can we also fix the comments in this file

Comment (pan3793) [reply build/dist:148]: This PR has been split to several small PRs

Comment (pan3793) [reply build/dist:148]: Please move discussion to #889

File: build/release/release.sh [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+#
+
+set -o pipefail
+set -e
+set -x
+
+ASF_USERNAME=${ASF_USERNAME:?"ASF_USERNAME is required"}
+ASF_PASSWORD=${ASF_PASSWORD:?"ASF_PASSWORD is required"}
+RELEASE_VERSION=${RELEASE_VERSION:?"RELEASE_VERSION is required"}
+RELEASE_RC_NO=${RELEASE_RC_NO:?"RELEASE_RC_NO is required"}
```
Comment (yaooqinn) [inline build/release/release.sh:27]: is this optional?

Comment (pan3793) [reply build/release/release.sh:27]: no, all these four are required

**Review by yaooqinn — APPROVED** [review]

**Pull request closed without merging**